File: src/main/resources/static/feng/purchase/demand/demand.js
var prefix = "/purchase/demand"

$(function() {
	var columns = [{
            checkbox: true
        },
        {
            field: 'demandId',
            title: 'Demand Id'
        },
        {
            field: 'demandDate',
            title: 'Demand Date'
        },
        {
            field: 'vendorId',
            title: 'Vendor Id'
        },
        {
            field: 'vendorName',
            title: 'Vendor Name'
        },
        {
            field: 'itemCode',
            title: 'Item Code'
        },
        {
            field: 'itemName',
            title: 'Item Name'
        },
        {
            field: 'price',
            title: 'Price'
        },
        {
            field: 'quantity',
            title: 'Quantity'
        },
        {
            field: 'urgencyDegree',
            title: 'Urgency Degree'
        },
        {
            field: 'createBy',
            title: 'Create By'
        },
        {
            field: 'createTime',
            title: 'Create Time'
        },
        {
            title: 'Action',
            align: 'center',
            formatter: function(value, row, index) {
            	var actions = [];
				actions.push('<a class="btn btn-primary btn-sm" href="#" title="Edit" mce_href="#" onclick="edit(\'' + row.demandId + '\')"><i class="fa fa-edit"></i></a> ');
				actions.push('<a class="btn btn-warning btn-sm" href="#" title="Remove" onclick="remove(\'' + row.demandId + '\')"><i class="fa fa-remove"></i></a>');
				return actions.join('');
            }
        }];
	var url = prefix + "/list";
	$.initTable(columns, url);
});

/*Demand Management-新增*/
function add() {
    var url = prefix + '/add';
    layer_show("add Demand", url, '800', '600');
}

/*Demand Management-修改*/
function edit(demandId) {
    var url = prefix + '/edit/' + demandId;
    layer_show("edit Demand", url, '800', '600');
}

// 单条删除
function remove(id) {
	$.modalConfirm("Do you want to remove this item？", function(r) {
		_ajax(prefix + "/remove/" + id, "", "post", r);
    })
}

// 批量删除
function batchRemove() {
	var rows = $.getSelections("demandId");
	if (rows.length == 0) {
		$.modalMsg("Please select the data you want to remove ", "warning");
		return;
	}
	$.modalConfirm("Do you want to remove " + rows.length + " demands?", function(r) {
		_ajax(prefix + '/batchRemove', { "ids": rows }, "post", r);
	});
}

File: src/main/java/com/feng/project/purchase/demand/domain/Demand.java
package com.feng.project.purchase.demand.domain;

import com.feng.framework.web.page.PageDomain;
import com.feng.project.product.production.domain.Production;
import com.feng.project.system.dept.domain.Dept;
import lombok.Data;
import lombok.EqualsAndHashCode;

/**
 * 供应商对象 pur_demand
 * pur_demand
 * @author feng
 */
@Data
@EqualsAndHashCode(callSuper = true)
public class Demand extends PageDomain
{
    /** 供应商ID */
    private Long demandId;
    /**  */
    private String demandDate;
    /**  */
    private String vendorId;
    /**  */
    private String vendorName;
    /**  */
    private String itemCode;
    /**  */
    private Double price;
    /**  */
    private Double quantity;
    /**  */
    private String urgencyDegree;
    /** 状态（0正常 1停用） */
    private int status;
    /** 创建者 */
    private String createBy;
    /** 创建时间 */
    private String createTime;
    /** 更新时间 */
    private String updateTime;
    /** 更新者 */
    private String updateBy;
    /** 备注 */
    private String remark;
    /**  */
    private String specification;
    /**  */
    private String itemName;
    /** 产品对象 */
    private Production production;

}
